cli/azd/pkg/auth/manager.go

```diff
@@ -122,6 +125,19 @@ func EnsureLoggedInCredential(ctx context.Context, credential azcore.TokenCreden
 	return &token, nil
 }
 
+func rebuildMsalClient(clientOptions []public.Option, tenantId string) (public.Client, error) {
```

Comment:
It seems like we have enough places where we need to build a different client. Do you think we should avoid holding to a default client on construction? It avoids all this confusion about precreated client vs. overridden client